File: J9Ch/src/S_6_Dynamic_Problem_II/Optional_11/python_cpp/_562_Backpack_IV_Medium.cpp

class Solution {
public:
    /**
     * @param nums an integer array and all positive numbers, no duplicates
     * @param target an integer
     * @return an integer
     */
    int backPackIV(vector<int>& nums, int target) {
        // Write your code here
        vector<int> dp(target + 1);
        dp[0] = 1;
        for (auto a : nums) {
            for (int i = a; i <= target; ++i) {
                dp[i] += dp[i - a];
            }
        }
        return dp.back();
    }
};

File: J9Ch/src/S_6_Dynamic_Problem_II/Optional_11/python_cpp/_562_Backpack_IV_Medium.py

class Solution:
    # @param {int[]} nums an integer array and all positive numbers, no duplicates
    # @param {int} target an integer
    # @return {int} an integer
    def backPackIV(self, nums, target):
        # Write your code here
        dp = [0 for i in xrange(target + 1)]
        dp[0] = 1

        for a in nums:
            for j in xrange(a, target+1):
                dp[j] += dp[j - a]

        return dp[target]